### PR TITLE
tagging: check for possible null tag to avoid segmentation violation.

### PR DESCRIPTION
**Pull request by GrahamByrnes**
Could fix #4400.

### CODE REVIEW

File: src/libs/tagging.c [hunk 1]
```diff
@@ -2353,10 +2353,12 @@ static gint sort_tree_count_func(GtkTreeModel *model, GtkTreeIter *a, GtkTreeIte
 
 static gint sort_tree_tag_func(GtkTreeModel *model, GtkTreeIter *a, GtkTreeIter *b, dt_lib_module_t *self)
 {
-  char *tag_a = 0;
-  char *tag_b = 0;
```
Comment (phweyland) [inline src/libs/tagging.c]: ?? !!!

**Comment by phweyland** [comment]
The trace is clear. But I'm afraid this should not happen and this fix avoids the seg fault but we don't resolve the real cause.
However, so far I haven't the answer. I've tried (from sqlitestudio):
- to attach a non existing tag on an image. Hovering on it doesn't crash.
- to attach a NULL tag on an image. I've another crash in db_sanitize at launch time (normal)
- to attach an empty tag ("") on an image. Hovering on it show the empty tag.

In the backtrace, the crash is related to the display of attached tags. That means that either we have some selected image(s) or some image is hovered.

**Comment by GrahamByrnes** [comment]
Surely, yes. Here are 3 from a crash yesterday. Note that the first is the only one with any mods.
The second file contains the same files from when I ran after the older version, so no crash was involved.

[Test4xmp.zip](https://github.com/GrahamByrnes/darktable/files/4289522/Test4xmp.zip)
[Test5xmp.zip](https://github.com/GrahamByrnes/darktable/files/4289524/Test5xmp.zip)

**Comment by GrahamByrnes** [comment]
BTW, I don't know my arse from my elbow in sql, but is that double-quote at the end of the first line correct?
(common/database.c line 1698)

sqlite3_exec(db->handle, "CREATE TABLE main.history_hash (imgid INTEGER PRIMARY KEY, "
               "basic_hash BLOB, auto_hash BLOB, current_hash BLOB)",
               NULL, NULL, NULL);

**Comment by phweyland** [comment]
> is that double-quote at the end of the first line correct?

yes

**Comment by phweyland** [comment]
I've built dt on windows 10 and I can reproduce your issues...

**Comment by phweyland** [comment]
I'm happy to develop on linux now (windows is so slow compared to linux ...).

With #4426 the crashes have gone from windows (I don't understand why it doesn't crash on linux, maybe a different implementation of malloc()).

I would remove the tests for null tags of this PR (I think that's a consequence of the other bug ( #4426 )).

**Pull request merged**